### DISCUSSION TITLE
Suggested upgrades to React canary versions are downgrades instead

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [renovatebot/renovate#32101] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I am using Next v15 which relies on specific canary versions of React but it seems those don't follow proper SemVer conventions and thus Renovate suggests downgrades instead. At the time of writing this an example PR is the following (which might change and I'm attaching a screenshot of it as well): https://github.com/ChristianIvicevic/adios-axios/pull/52

![CleanShot 2024-10-23 at 10 57 01@2x](https://github.com/user-attachments/assets/a5a2e9ff-3dd6-47eb-a337-05eb29ebe28b)

At the time of writing this the actual version that is up to date is `69d4b800-20241021` instead.

It might be worthwhile to evaluate whether to handle React upgrades of canaries in a different way than just relying on SemVer which seems to be reason that this is happening.

### Logs (if relevant)

<details><summary>Logs</summary>

```

n/a

```

</details>

**Comment by rarkins** [comment]
Adding to renovate's workarounds presets would be appropriate here
